### PR TITLE
Remove to device in network

### PR DESCRIPTION
**Pull request by henryyang42**
Reproducibility is ensured in https://github.com/ASUS-AICS/LibMultiLabel/tree/reproduce

### CODE REVIEW

**Comment by Eleven1Liu** [comment]
@henryyang42 I just test CAML/MIMIC-50 without `.to(device)`. It won't affect to results.

**Review by Eleven1Liu — APPROVED** [review]

**Comment by henryyang42** [comment]
@Eleven1Liu I got different result on kim_cnn/rcv1. If CAML&CNN on MIMIC-50 dataset without ``.to(device)``  can produce same result. We may need to go deeper on random seeds.

Previously we didn't set seed on  ``torch.cuda.manual_seed_all(seed)`` and ``random.seed``. Now we use PL's ``seed_everything``, which handles 
```python
    random.seed(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed_all(seed)
```

**Comment by Eleven1Liu** [comment]
> @Eleven1Liu I got different result on kim_cnn/rcv1. If CAML&CNN on MIMIC-50 dataset without `.to(device)` can produce same result, we may need to go deeper on random seeds (One difference between rcv1 and MIMIC is that rcv1 does not have a predefined validation set, I am not sure if `train_test_split(datasets['train'], test_size=config.val_size, random_state=42)` will cause any issue if config.seed != 42. Another difference is that CAML will not apply `networks.get_init_weight_func`).
> 
> Previously we didn't set seed on `torch.cuda.manual_seed_all(seed)` and `random.seed(seed)`. Now we use PL's `seed_everything`, which handles
> 
> ```python
>     random.seed(seed)
>     np.random.seed(seed)
>     torch.manual_seed(seed)
>     torch.cuda.manual_seed_all(seed)
> ```

True. It's different ...
